scopes/semantics/schema/schema.main.runtime.ts

```diff
@@ -77,12 +84,48 @@ export class SchemaMain {
    */
   async getSchema(component: Component): Promise<APISchema> {
     this.logger.debug(`getSchema of ${component.id.toString()}`);
-    const env = this.envs.getEnv(component).env;
-    if (typeof env.getSchemaExtractor === 'undefined') {
-      throw new Error(`No SchemaExtractor defined for ${env.name}`);
+
+    // if on workspace get schema from ts server
+    if (this.workspace) {
```

Comment:
If this is not an expensive op it might be better to calculate it on load

Reply:
it is as expensive as compiling the component
Not sure what happens onComponentLoad and where this should be persisted and then fetched from when eventually requested through the runtime's API
I do see some code commented out in the provider of the `SchemaMainRuntime` which tried to fetch the component schema. 
Will connect with David/Ran to understand it better